app/hooks/forms.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useState} from 'react';
+
+export type FormInput = {
+    value: string;
+    onChange: (value: string) => void;
+}
+
+export const useFormInput = (initialValue?: string): FormInput => {
+    const [value, setValue] = useState<string>(initialValue || '');
+
+    function handleChange(text: string) {
```

Comment:
Why create this function here, instead of passing setValue? Not 100% sure, but setValue should be "constant" (always the same function) but creating here the handleChange function means that every call of `useFormInput` returns a different version of `onChange`. (i.e. `"onChange on render 1" !== "onChange on render 2"`)

Reply:
The main intention was to reduce the number of props passed to the `edit_channel_info` component.

From 
```
onDisplayNameChange={setDisplayName}
onHeaderChange={setHeader}
onPurposeChange={setPurpose}
displayName={displayName}
header={header}
purpose={purpose}
```

To
```
displayName={displayName}
header={header}
purpose={purpose}
```

Reply:
With 2 dissenting opinions, I will change remove this idea and pass in both value and setvalue as props.